SPARCore/GADArticle.m

```diff
@@ -1,104 +1,99 @@
-    }
+    NSMutableArray <GADArticle *> *articles = [NSMutableArray new];
     
     for (NSDictionary *element in jsonArray) {
-        //Map fields of element to fields of article
-        GADArticle *article = [[GADArticle alloc] init];
-        //datePublished field is a UNIX Timestamp number - converting to NSDate here
-        int timeStamp = (int)element[API_DATE_PUBLISHED];
-        article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
-        article.brief = element[API_BRIEF];
-        article.headerImage = element[API_HEADER_IMAGE];
-        article.publicationId = element[API_PUBLICATION_ID];
-        article.articleId = element[API_ARTICLE_ID];
-        article.title = element[API_TITLE];
-        if (element[API_CONTENT]) {
-            article.content = element[API_CONTENT];
-            article.authors = element[API_AUTHORS];
-        }
+        GADArticle *article = [self articleFromDictionary:element];
         [articles addObject:article];
     }
     return articles;
 }
 
-+ (NSURL *) baseURL {
-    NSURL *queryURL = [NSURL URLWithString:API_HOSTNAME];
-    queryURL = [NSURL URLWithString:API_PREFIX relativeToURL:queryURL];
-    return queryURL;
++ (GADArticle *) articleFromDictionary: (NSDictionary*)dict {
+    //Map fields of element to fields of article
+    GADArticle *article = [[GADArticle alloc] init];
+    //datePublished field is a UNIX Timestamp number - converting to NSDate here
+    int timeStamp = (int)dict[API_KEY_DATE_PUBLISHED];
+    article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
+    article.headerImage = [NSURL URLWithString:dict[API_KEY_HEADER_IMAGE]];
+    article.publicationId = dict[API_KEY_PUBLICATION_ID];
+    article.articleId = dict[API_KEY_ARTICLE_ID];
+    article.title = dict[API_KEY_TITLE];
+    article.authors = dict[API_KEY_AUTHORS];
+    
+    return article;
 }
 
-+ (NSURL *) urlForArticlesFromPublication: (NSString *)publicationId{
-    NSURL *queryURL = [GADArticle baseURL];
-    queryURL = [NSURL URLWithString:API_PUBLICATION_PATH relativeToURL:queryURL];
-    queryURL = [NSURL URLWithString:publicationId relativeToURL:queryURL];
-    queryURL = [NSURL URLWithString:API_ARTICLE_PATH relativeToURL:queryURL];
-    return queryURL;
+- (void) fetchFullTextWithCompletion: (void(^_Nonnull)(GADArticle *_Nullable article,
+                                                       NSError *_Nullable error))completion {
+    NSURL *queryURL = [self urlForFullArticle];
+    
+    NSMutableURLRequest *req = [NSMutableURLRequest requestWithURL:queryURL cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:timeoutInterval];
+    [req setHTTPMethod:@"GET"];
+    [req setValue:@"application/json" forHTTPHeaderField: @"Content-Type"];
+    
+    NSURLSessionDataTask *task = [[NSURLSession sharedSession]dataTaskWithRequest:req completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
```

Comment:
Same, please keep lines under 100 characters. Xcode can put a guideline in your GUI: ` Xcode -> Preferences -> Text Editing`; check `Page guide at column: 100`.